src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/slack/impl/SlackMessagingServiceImpl.java

```diff
@@ -0,0 +1,81 @@
+package uk.gov.companieshouse.chipsrestinterfacesconsumer.slack.impl;
+
+import com.slack.api.Slack;
+import com.slack.api.methods.MethodsClient;
+import com.slack.api.methods.SlackApiException;
+import com.slack.api.methods.request.chat.ChatPostMessageRequest;
+import com.slack.api.methods.response.chat.ChatPostMessageResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.common.ApplicationLogger;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.slack.SlackMessagingService;
+
+import java.io.IOException;
+import java.util.List;
+
+@Service
+public class SlackMessagingServiceImpl implements SlackMessagingService {
+
+    public static final int LIMIT = 20;
+    private final ApplicationLogger logger;
+
+    @Value("${SLACK_CHANNEL}")
+    private String slackChannel;
+
+    @Value("${SLACK_ACCESS_TOKEN}")
+    private String slackAccessToken;
+
+    @Value("${RUN_APP_IN_ERROR_MODE}")
+    private boolean inErrorMode;
+
+    private MethodsClient methods;
```

Comment:
for thread safety, i think you can move the methods var inside the method

Reply:
Problem is it would be very hard or oeven impossible to unit test if we do.

Reply:
Not sure what to do about this we may need an alternative solution if we want to keep the new tests.

Reply:
Moved the testable code into a separate method to spy on.